### PR TITLE
Add expire time for redis keys

### PR DESCRIPTION
**Pull request by maximemenager**

### CODE REVIEW

File: cozy.example.yaml [hunk 1]
```diff
@@ -205,6 +205,7 @@ redis:
     konnectors: 5
     realtime: 6
     log: 7
+    rate_limiting: 8
```
Comment (nono) [inline cozy.example.yaml]: @sblaisot we will need to update the config file for this pull-request

Comment (sblaisot) [reply cozy.example.yaml]: no problem in our infra.
ping @aeris for self-hosted

File: web/auth/rate_limiting.go [hunk 2]
```diff
@@ -98,7 +98,14 @@ func NewRedisCounter(client redis.UniversalClient) Counter {
 }
 
 func (r *redisCounter) Increment(key string, timeLimit time.Duration) (int64, error) {
-	return r.Client.Incr(key).Result()
+	count, err := r.Client.Incr(key).Result()
+	if err != nil {
+		return 0, err
+	}
+	if count == 1 {
```
Comment (sblaisot) [inline web/auth/rate_limiting.go]: Why do you count time from first failed password and not from last failed password ?

**Comment by nono** [comment]
For the self-hosted, there is nothing to do: we use an in-memory version in the stack, not redis.

File: web/auth/rate_limiting.go [hunk 3]
```diff
@@ -98,7 +98,14 @@ func NewRedisCounter(client redis.UniversalClient) Counter {
...
+		r.Client.Expire(key, timeLimit)
```
Comment (edas) [inline web/auth/rate_limiting.go]: Non blocking : You probably should have used Redis transactions to deal with concurrency. You may have 1000 requests done by another client between L101 and L106 here

Comment (nono) [reply web/auth/rate_limiting.go]: I don't see the concurrency issue here. Can you explain why it matters if 1000 requests have been made between L101 and L106?

Comment (edas) [reply web/auth/rate_limiting.go]: It's not an issue (so non blocking) but probably a bad pattern to not use a transaction. No real consequence here

Comment (nono) [reply web/auth/rate_limiting.go]: And, even if we have a race condition, I don't think we can do what we want with a redis transaction (neither multi/exec nor watch/discard), but there is still the lua script way.

Comment (nono) [reply web/auth/rate_limiting.go]: > probably a bad pattern to not use a transaction

I disagree: it's true for most databases, but not redis. Redis is single-threaded and its transactions are not comparable to transactions in SQL world for example.

**Pull request merged**